profile-widget/profilewidget2.cpp

```diff
@@ -2094,6 +2101,42 @@ bool ProfileWidget2::PictureEntry::operator< (const PictureEntry &e) const
 	return std::tie(offset.seconds, filename) < std::tie(e.offset.seconds, e.filename);
 }
 
+// Calculate the y-coordinates of the thumbnails, which are supposed to be sorted by x-coordinate.
+// This will also change the order in which the thumbnails are painted, to avoid weird effects,
+// when items are added later to the scene. This is simply done by increasing the Z-value.
+void ProfileWidget2::calculatePictureYPositions()
+{
+	double lastX = -1.0, lastY;
+	double z = 0.0;
+	for (PictureEntry &e: pictures) {
+		if (!e.thumbnail)
+			continue;
+		// let's put the picture at the correct time, but at a fixed "depth" on the profile
+		// not sure this is ideal, but it seems to look right.
+		double x = e.thumbnail->x();
+		double y;
+		if (lastX < 0.0)
+			y = 10;
+		else if (fabs(x - lastX) < 3 && lastY <= (10 + 14 * 3))
+			y = lastY + 3;
+		else
+			y = 10;
```

Comment:
i think can be written as:
```
if (fabs(x - lastX) < 3 && lastY <= (10 + 14 * 3))
	y = lastY + 3;
else
	y = 10;
```

Reply:
This is only copy & paste, but yeah - let's just simplify this.